react/features/facial-recognition/actions.js

```diff
@@ -52,80 +57,99 @@ let duplicateConsecutiveExpressions = 0;
 
         const workerBlob = new Blob([ `importScripts("${workerUrl}");` ], { type: 'application/javascript' });
 
         workerUrl = window.URL.createObjectURL(workerBlob);
-        worker = new Worker(workerUrl, { name: 'Facial Expression Worker' });
+        worker = new Worker(workerUrl, { name: 'Face Recognition Worker' });
         worker.onmessage = function(e: Object) {
-            const { type, value } = e.data;
+            const { faceExpression, faceBox } = e.data;
 
-            // receives a message indicating what type of backend tfjs decided to use.
-            // it is received after as a response to the first message sent to the worker.
-            if (type === INTERVAL_MESSAGE) {
-                value && dispatch(setDetectionTimeInterval(value));
-            }
-
-            // receives a message with the predicted facial expression.
-            if (type === FACIAL_EXPRESSION_MESSAGE) {
-                sendDataToWorker(worker, imageCapture);
-                if (!value) {
-                    return;
-                }
-                if (value === lastFacialExpression) {
+            if (faceExpression) {
+                if (faceExpression === lastFacialExpression) {
                     duplicateConsecutiveExpressions++;
                 } else {
                     if (lastFacialExpression && lastFacialExpressionTimestamp) {
-                        dispatch(
-                        addFacialExpression(
+                        dispatch(addFacialExpression(
                             lastFacialExpression,
                             duplicateConsecutiveExpressions + 1,
                             lastFacialExpressionTimestamp
-                        )
-                        );
+                        ));
                     }
-                    lastFacialExpression = value;
+                    lastFacialExpression = faceExpression;
                     lastFacialExpressionTimestamp = Date.now();
                     duplicateConsecutiveExpressions = 0;
                 }
             }
+
+            if (faceBox) {
+                const state = getState();
+                const conference = getCurrentConference(state);
+                const localParticipant = getLocalParticipant(state);
+
+                if (getParticipantCount(state) > 1) {
+                    sendFaceBoxToParticipants(conference, faceBox);
+                }
+
+                dispatch({
+                    type: UPDATE_FACE_COORDINATES,
+                    faceBox,
+                    id: localParticipant.id
+                });
+            }
         };
+
+        const { enableFacialRecognition, faceCoordinatesSharing } = getState()['features/base/config'];
+        const detectionTypes = [
+            faceCoordinatesSharing?.enabled && DETECTION_TYPES.FACE_BOX,
+            enableFacialRecognition && DETECTION_TYPES.FACE_EXPRESSIONS
+        ].filter(Boolean);
```

Comment:
Do we still want to initialize the worker if both are disabled and the array is empty?

Reply:
the check for initializing the worker is done in the middleware, this is just for passing the array, at this point we know that at least one of them is active